software/winrm/src/main/java/org/apache/brooklyn/location/winrm/WinRmMachineLocation.java

```diff
@@ -197,7 +198,15 @@ public String getHostname() {
     @Nullable
     protected String getHostAndPort() {
         String host = getHostname();
-        return (host == null) ? null : host + ":" + config().get(WINRM_PORT);
+        return (host == null) ? null : host + ":" + (config().get(USE_SECURE_WINRM) ? 5986 : 5985);
+    }
+
+    public int getPort() {
+        if (getConfig(WINRM_PORT) != null) {
+            return getConfig(WINRM_PORT);
+        } else {
+            return getConfig(USE_SECURE_WINRM) ? 5986 : 5985;
```

Comment:
It seems very strange to have `VanillaWindowsProcess.WINRM_PORT` and also a completely separate `WINRM_PORT` config on the location. I see that in a `VanillaWindowsProcessImpl.preStart` it reads the `machine. USE_SECURE_WINRM` but it then makes its own decision about the port number, rather than using the value from the machine. I can see how this works because we're using standard port numbers, but it's confusing (and would likely get very confusing if non-standard port numbers were every used).